soft4pes/model/conv/inverter.py

```diff
@@ -0,0 +1,100 @@
+""" 
+n-level inverter model.
+"""
+
+from itertools import product
+import numpy as np
+
+
+class Inverter:
+    """ 
+    Class representing an 2- or 3-level inverter with constant dc-link voltage. 
+
+    Attributes
+    ----------
+    v_dc : float
+        Dc_link voltage [pu]
+    nl : float
+        Number of voltage levels in the inverter
+    SW_COMB : 3^nl x 3 ndarray of ints
+        Possible converter switching states.
+    """
+
+    def __init__(self, v_dc, nl, base):
+        """
+        Initialize an Inverter instance.
+
+        Parameters
+        ----------
+        v_dc : float
+            Dc_link voltage [V].
+        nl : int
+            Number of voltage levels in the inverter.
+        base : base value object
+            Base values.
+        """
+
+        self.v_dc = v_dc / base.V
+        self.nl = nl
+
+        if nl == 2:
```

Comment:
From this point onwards, the code is designed while having in mind as sort of direct control where we enumerate the switch positions. It is not wrong, but it might not be the best file to have this. Maybe in the future this will have to change and have a dedicated file for this.

Reply:
I agree, things related to direct control should be moved elsewhere in the future